include/libtorrent/packet_pool.hpp

```diff
@@ -78,78 +78,82 @@ namespace libtorrent
 		std::uint8_t buf[1];
 	};
 
+
+	struct packet_deleter
+	{
+		void operator()(packet* handle) const
+		{
+			std::free(handle);
```

Comment:
awesome. one more thing I forgot to mention: would you mind also destructing the packet object?
``handle->~packet();`` just before freeing it